code/controlconfig/controlsconfig.h

```diff
@@ -62,7 +62,7 @@ typedef struct config_item {
 	short key_default;		//!< default key bound to action
 	short joy_default;		//!< default joystick button bound to action
 	char tab;				//!< what tab (category) it belongs in
-	bool hasXSTR;			//!< whether we should translate this with an XSTR
+	int hasXSTR = 0;		//!< what string index we should use to translate this with an XSTR
```

Comment:
Try this.  Take the zero assignment out of the struct and put it right before the `if (optional_string("$Has XStr:")) {` parsed line.

Reply:
Will do, thanks!

Reply:
Also, please either update the name of the field so that it doesn't imply a boolean value anymore or use two fields, one for determining whether the item has an XSTR and a second one for specifying the index.